sdks/go/pkg/beam/core/funcx/fn.go

```diff
@@ -374,6 +390,8 @@ func New(fn reflectx.Func) (*Fn, error) {
 			kind = FnWindow
 		case t == typex.BundleFinalizationType:
 			kind = FnBundleFinalization
+		case t == state.ProviderType:
```

Comment:
Should we have the types for state and timers defined in typex package? I was thinking this for timers as well.

Reply:
I'd probably vote we leave it as is unless there's a reason to switch - the [typex package generally only exports types for structs it defines](https://github.com/apache/beam/blob/48bad7d966a583055669850eb9fb558782f636a8/sdks/go/pkg/beam/core/typex/special.go#L45) and IMO it makes sense to stay it in the state package to keep consistent with that. I don't feel super strongly though (@lostluck might have opinions too)